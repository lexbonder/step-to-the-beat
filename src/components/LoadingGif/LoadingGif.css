.LoadingGif {
  margin: 20vh auto;
  text-align: center;
}

.LoadingGif img,
.LoadingGif {
  width: 300px;
}

.LoadingGif h1 {
  color: #4E9AF5;
  margin-bottom: 10px;
}