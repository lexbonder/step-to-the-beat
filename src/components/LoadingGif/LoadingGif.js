import React from 'react';
import bacon from '../../loadingGifs/bacon-running.gif';
import bears from '../../loadingGifs/bears-running.gif';
import connie from '../../loadingGifs/connie-running.gif';
import gingerbreadGirl from '../../loadingGifs/gingerbread-girl-running.gif';
import goku from '../../loadingGifs/goku-running.gif';
import pikachu from '../../loadingGifs/pikachu-running.gif';
import roadrunner from '../../loadingGifs/roadrunner-running.gif';
import sonic from '../../loadingGifs/sonic-running.gif';
import spongeBob from '../../loadingGifs/spongebob-running.gif';
import './LoadingGif.css';

const LoadingGif = () => {
  const gifs = [bacon, bears, connie, gingerbreadGirl,
    goku, pikachu, roadrunner, sonic, spongeBob
  ];
  const randomGif = gifs[Math.floor(Math.random() * gifs.length)];
  return (
    <div className='LoadingGif'>
      <h1>Loading...</h1>
      <img src={randomGif} alt='loading gif' />
    </div>
  );
};

export default LoadingGif;