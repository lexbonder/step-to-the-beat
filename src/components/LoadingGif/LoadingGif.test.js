import React from 'react';
import { shallow } from 'enzyme';
import LoadingGif from './LoadingGif';

describe('LoadingGif', () => {
  it('should match the snapshot', () => {
    Math.random = jest.fn().mockImplementation(() => 0)
    const wrapper = shallow(<LoadingGif />);
    expect(wrapper).toMatchSnapshot();
  });
});