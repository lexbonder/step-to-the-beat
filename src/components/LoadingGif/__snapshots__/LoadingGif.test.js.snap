// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingGif should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoadingGif />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Loading...
</h1>,
        <img
          alt="loading gif"
          src="bacon-running.gif"
/>,
      ],
      "className": "LoadingGif",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Loading...",
        },
        "ref": null,
        "rendered": "Loading...",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "alt": "loading gif",
          "src": "bacon-running.gif",
        },
        "ref": null,
        "rendered": null,
        "type": "img",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Loading...
</h1>,
          <img
            alt="loading gif"
            src="bacon-running.gif"
/>,
        ],
        "className": "LoadingGif",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Loading...",
          },
          "ref": null,
          "rendered": "Loading...",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "loading gif",
            "src": "bacon-running.gif",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
